### PR TITLE
refactor(ekf_localizer): isolate measurement models as functions

### PR DESCRIPTION
**Pull request by IshitaTakeshi**
## Description

Isolate the parts that create measurement matrices and covariances to functions. 
 

## Tests performed

Basic smoke tests that set brief inputs and observe the results. 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: localization/ekf_localizer/include/ekf_localizer/pose_measurement.hpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Copyright 2022 Autoware Foundation
```
Comment (kminoda) [inline localization/ekf_localizer/include/ekf_localizer/pose_measurement.hpp:1]: Do we need this script? (Just forgot to remove?)

Comment (IshitaTakeshi) [reply localization/ekf_localizer/include/ekf_localizer/pose_measurement.hpp:1]: I think so, as in other files

Comment (kminoda) [reply localization/ekf_localizer/include/ekf_localizer/pose_measurement.hpp:1]: Sorry for the ambiguity.
What I meant was about the file (=`pose_measurement.hpp`) itself.

Comment (IshitaTakeshi) [reply localization/ekf_localizer/include/ekf_localizer/pose_measurement.hpp:1]: I forgot to remove it. Thank you for notifying

Comment (IshitaTakeshi) [reply localization/ekf_localizer/include/ekf_localizer/pose_measurement.hpp]: Resolved in https://github.com/autowarefoundation/autoware.universe/pull/1795/commits/bdcb3fed2155bc48a60f5e24938ec5274e6c46db

File: localization/ekf_localizer/src/measurement.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+Eigen::Matrix<double, 2, 6> twistMeasurementMatrix()
+{
+  Eigen::Matrix<double, 2, 6> C = Eigen::Matrix<double, 2, 6>::Zero();
+  C(0, IDX::VX) = 1.0;  // for vx
+  C(1, IDX::WZ) = 1.0;  // for wz
+  return C;
+}
+
+Eigen::Matrix3d poseMeasurementCovariance(
+  const std::array<double, 36ul> & covariance, const double smoothing_step)
```
Comment (kminoda) [inline localization/ekf_localizer/src/measurement.cpp:37]:
```suggestion
Eigen::Matrix3d poseMeasurementCovariance(
  const std::array<double, 36ul> & covariance, const int smoothing_step)
```
It's better to accept only integer as `smoothing_step`.
(same applies to `twistMeasurementCovariance` too)

Comment (IshitaTakeshi) [reply localization/ekf_localizer/src/measurement.cpp:37]: If it expects only an unsigned integer, I'd suggest size_t and cast the type inside of the function.

Comment (IshitaTakeshi) [reply localization/ekf_localizer/src/measurement.cpp]: Resolved in https://github.com/autowarefoundation/autoware.universe/pull/1795/commits/87de9beb3a3f3e648be837f970c12a0b99240efd

I also added compile options

**Review by kminoda — APPROVED** [review]
LGTM

**Pull request merged**